mautic/mautic
# Always track utm tags from page url

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | staging
| Bug fix?                               | no
| New feature?                           | yes
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | yes
| Related user documentation PR URL      | https://github.com/mautic/mautic-documentation/pull/128
| Related developer documentation PR URL | 
| Issue(s) addressed                     | 

#### Description:
Related to https://forum.mautic.org/t/utm-tag-contact-properties-and-reporting/10320/5

This PR adds the functionality to record UTM tags for all requests. Including via Javascript (/mtc/event).

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Create a landing page
3. Open the landing page in incognito mode with utm tags attached to the url. E.g. ?utm_source=linkedin&utm_medium=social&utm_campaign=mautic&utm_content=testing
4. Check the contact history to see if UTM tags are recorded for the anonymous contact. 

Note that UTM tags are only recorded the first time a user visits a certain page

If you can test also the js tracking:

```
curl --location --request POST 'http://domain.ddev.site/index_dev.php/mtc/event' \
--header 'Content-Type: application/json' \
--data-raw '{
    "page_title": "Testpage",
    "page_language": "en-GB",
    "page_referrer": "",
    "page_url": "https://www.domain.com/mautic/?utm_source=linkedin&utm_medium=social&utm_campaign=mautic&utm_content=test",
    "counter": 0,
    "mautic_device_id": "qowvkyd2r1232312okcg2qw",
    "resolution": "1792x1120",
    "timezone_offset": -120,
    "platform": "MacIntel",
    "do_not_track": 1,
    "adblock": false,
    "fingerprint": "fqwc25wa2d659c4123c7f5d5789012332"
}'

```
